src/video/SDL_blit.h

```diff
@@ -443,21 +437,58 @@ extern SDL_BlitFunc SDL_CalculateBlitA(SDL_Surface *surface);
         }                                             \
     }
 
+/* Convert any 32-bit 4-bpp pixel to ARGB format */
+#define PIXEL_TO_ARGB_PIXEL(src, srcfmt, dst)         \
```

Comment:
I'd recommend just checking the pixel format has the expected layout and using the pixel as-is. It should be possible to support both ARGB to ARGB and ABGR to ABGR blits this way, and it'll likely be faster to add a special case for ARGB to ABGR and ABGR to ARGB blits instead of using generic conversion code.